src/uwtools/set_config.py

```diff
@@ -122,5 +128,25 @@ def create_config_obj(argv):
             out_object = config_obj
         out_object.dump_file(user_args.outfile)
 
+    ##begin show_format
+    # if --show_format, print a list of required configuration settings to stdout
+
+    if user_args.show_format:
+        # first ensure all required args are present
+        assert(
+            user_args.show_format and user_args.outfile
+        ), "args: --show_format also requires -outfile for reference"
```

Comment:
`assert` should be reserved for tests and non "production" code (see [ref](https://towardsdatascience.com/2-caveats-to-avoid-when-using-python-assert-statement-8060e123c26a)).

Reply:
Replaced with `raise argparse.ArgumentError` which is much better, thanks!